api/sparse/urls.py

```diff
@@ -11,7 +11,7 @@
         name=views.SparseNodeList.view_name,
     ),
     url(
-        r'^nodes/(?P<node_id>\w+)/',
+        r'^nodes/(?P<node_id>\w+)/$',
```

Comment:
This and the route change on ln 40 are the only truly necessary changes for this ticket, the rest is just to write more organized tests.